include/mrsh/getopt.h

```diff
@@ -0,0 +1,8 @@
+#ifndef MRSH_GETOPT_H
+#define MRSH_GETOPT_H
+
+#include <getopt.h>
+
+int mrsh_getopt(int argc, char * const argv[], const char *optstring);
```

Comment:
Is it possible to add `mrsh_` prefixed variables for `optind` et al? I'd prefer not to mutate libc's state.

Reply:
That sounds sensible, I should be able to get that done tomorrow.

Reply:
I've just pushed an update which does this. I've also set `mrsh_optarg` to `NULL` at the start of each call.